src/corefx/System.Globalization.Native/pal_collation.c

```diff
@@ -239,29 +261,33 @@ UCollator* CloneCollatorWithOptions(const UCollator* pCollator, int32_t options,
     }
 
     UCollator* pClonedCollator;
-    std::vector<UChar> customRules = GetCustomRules(options, strength, isIgnoreSymbols);
-    if (customRules.empty())
+    UCharList* customRules = GetCustomRules(options, strength, isIgnoreSymbols);
+    if (customRules == NULL)
     {
-        pClonedCollator = ucol_safeClone(pCollator, nullptr, nullptr, pErr);
+        pClonedCollator = ucol_safeClone(pCollator, NULL, NULL, pErr);
     }
     else
     {
-        int32_t customRuleLength = customRules.size();
+        int32_t customRuleLength = customRules->size;
 
         int32_t localeRulesLength;
         const UChar* localeRules = ucol_getRules(pCollator, &localeRulesLength);
+        int32_t completeRulesLength = localeRulesLength + customRuleLength + 1;
+
+        UChar* completeRules = calloc(completeRulesLength, sizeof(UChar));
```

Comment:
>completeRules [](start = 15, length = 13)

we should check if the allocation succeeded after calling calloc

Reply:
OK.